packages/in_app_purchase/example/lib/main.dart

```diff
@@ -27,53 +27,115 @@ class _MyAppState extends State<MyApp> {
         appBar: AppBar(
           title: const Text('IAP Example'),
         ),
-        body: Center(
-            child: FutureBuilder<List<Widget>>(
-                future: buildStorefront(),
+        body: Column(
+          children: [
+            FutureBuilder(
+              future: _buildConnectionCheckTile(),
+              builder: (BuildContext context, AsyncSnapshot snapshot) {
+                if (!snapshot.hasData) {
+                  return Column(children: <Widget>[
+                    buildListCard(
+                        ListTile(title: const Text('Trying to connect...')))
+                  ]);
+                } else if (snapshot.error != null) {
+                  return Column(children: <Widget>[
+                    buildListCard(ListTile(
+                        title: Text(
+                            'Error connecting: ' + snapshot.error.toString())))
+                  ]);
+                }
+                return Column(
+                  children: snapshot.data,
+                );
+              },
+            ),
+            Expanded(
+              child: FutureBuilder(
+                future: InAppPurchaseConnection.instance.queryProductDetails(
+                    <String>[
+                  'consumable',
+                  'gas',
+                  'premium',
+                  'upgrade',
+                  'somethingNotValid'
```

Comment:
Ideally the example app should just be demoing the right way to use the API. I don't think it's worth deliberately checking for invalid product IDs here. All we're asking people to set up in the README so far is `consumable`, `upgrade`, and `subscription`.